test/tools/libtest.sh

```diff
@@ -468,42 +479,44 @@ test_tig()
 		export TIG_TRACE="$HOME/${prefix}tig-trace"
 	fi
 	touch -- "${prefix}stdin" "${prefix}stderr"
-	if [ -n "$debugger" ]; then
-		printf "*** Running tests in '%s/%s'\n" "$(pwd)" "$work_dir"
-		if [ -s "$work_dir/${prefix}stdin" ]; then
-			printf '*** - This test requires data to be injected via stdin.\n'
-			printf "***   The expected input file is: '%s'\n" "../${prefix}stdin"
-		fi
-		if [ "$#" -gt 0 ]; then
-			printf '*** - This test expects the following arguments: %s\n' "$*"
-		fi
-		(cd "$work_dir" && "$debugger" tig "$@")
-	else
-		set +e
-		runner=
-		# FIXME: Tell Valgrind to forward status code
-		if [ "$expected_status_code" = 0 ] && [ -n "$valgrind" ]; then
-			runner=valgrind_exec
-		fi
-		if [ -n "$runner" ]; then
-			if [ -s "${prefix}stdin" ]; then
-				(cd "$work_dir" && "$runner" tig "$@") < "${prefix}stdin" > "${prefix}stdout" 2> "${prefix}stderr.orig"
-			else
-				(cd "$work_dir" && "$runner" tig "$@") > "${prefix}stdout" 2> "${prefix}stderr.orig"
+	(
+		# subshell handles cleanup of cwd, variables, redirections, set +e
+		cd "$work_dir" || die "chdir failed"
+		if [ -n "$debugger" ]; then
+			printf "*** Running tests in '%s/%s'\n" "$HOME" "$work_dir"
+			if [ -s "$HOME/${prefix}stdin" ]; then
+				printf '*** - This test requires data to be injected via stdin.\n'
+				printf "***   The expected input file is: '%s'\n" "../${prefix}stdin"
+			fi
+			if [ "$#" -gt 0 ]; then
+				printf '*** - This test expects the following arguments: %s\n' "$*"
 			fi
+			"$debugger" tig "$@"
 		else
-			if [ -s "${prefix}stdin" ]; then
-				(cd "$work_dir" && tig "$@") < "${prefix}stdin" > "${prefix}stdout" 2> "${prefix}stderr.orig"
+			set +e
+			runner=exec
+			# FIXME: Tell Valgrind to forward status code
+			if [ "$expected_status_code" = 0 ] && [ -n "$valgrind" ]; then
+				runner=valgrind_exec
+				timeout=0
+			fi
+			if [ -s "$HOME/${prefix}stdin" ]; then
+				exec 4<"$HOME/${prefix}stdin"
+			elif [ -t 0 ] && [ -e /dev/tty ]; then
+				exec 4</dev/tty
```

Comment:
Since c883298355e02fc2c431cb68fe5edc212ec62fea this is not necessary since Tig opens `/dev/tty` by default.